utils/db-generator/query_db_generator.cpp

```diff
@@ -0,0 +1,1176 @@
+                        v += ", ";
+                    v += generate_one_value();
+                }
+                v += "]";
+                values.insert(v);
+            }
+            else
+            {
+                values.insert(generate_one_value());
+            }
+        }
+        return true;
+    }
+    void unify_type()
+    {
+        if (type & type::i)
+            type = type::i;
+        else if (type & type::f)
+            type = type::f;
+        else if (type & type::d)
+            type = type::d;
+        else if (type & type::dt)
+            type = type::dt;
+        else if (type & type::s)
+            type = type::s;
+        else if (type & type::b)
+            type = type::b;
+        else
+            throw "error determination column type " + name.first + '.' + name.second;
+    }
+};
+
+
+std::set<std::vector<std::string>>
+decart_mul(
+        std::set<std::vector<std::string>> & prev,
+        std::set<std::string> &              mul)
+{
+    std::set<std::vector<std::string>> result;
+    for (auto v : prev)
+        for (auto m : mul)
+        {
+            std::vector<std::string> tmp = v;
+            tmp.push_back(m);
+            result.insert(tmp);
+        }
+    return result;
+}
+
+
+class Table
+{
+public:
+    Table() {}
+
+    Table(std::string table_name)
+            : name(table_name) {}
+
+    std::string name;
+    std::set<std::string> columns;
+    std::map<std::string, Column> column_description;
+    bool column_exists(const std::string & column_name) const
+    {
+        return columns.count(column_name); // || columns_maybe.count(column_name);
+    }
+    void add_column(const std::string & column_name)
+    {
+        columns.insert(column_name);
+    }
+    void set_description(Column other)
+    {
+        column_description[other.name.second].merge(other);
+    }
+    void print()
+    {
+        std::cout << "Table\n";
+        std::cout << name << "\n";
+        std::cout << "Columns:\n\n";
+        for (auto column : columns)
+        {
+            std::cout << column << "\n";
+            if (column_description.count(column))
+                column_description[column].print();
+            std::cout << "\n";
+        }
+        std::cout << "\n";
+    }
+    void merge(Table other)
+    {
+        name = other.name;
+        columns.insert(other.columns.begin(), other.columns.end());
+        for (auto desc : other.column_description)
+            column_description[desc.first].merge(desc.second);
+    }
+    std::string create_query()
+    {
+        std::string create;
+        std::string db, _;
+        std::tie(db, _) = get_table_a_column(name);
+        create = "CREATE DATABASE IF NOT EXISTS " + db + ";\n\n";
+        create += "CREATE TABLE IF NOT EXISTS " + name + " (\n";
+        for (auto column = columns.begin(); column != columns.end(); ++column)
+        {
+            if (column != columns.begin())
+                create += ", -s\n";
```

Comment:
Why do you need it here? It will lead to syntax error.

Reply:
if you are talking about "-s", it is a mistake. 
i will remove it